examples/images_contours_and_fields/colorbar_annotation_demo.py

```diff
@@ -0,0 +1,54 @@
+"""
+========================
+Colorbar Annotation Demo
+========================
+
+Demonstrates how one can annotate the colorbar without needing to also add
+lines to a contour plot.
+
+Note that this is not a typical use case. The more common use case for
+annotations is to add lines to both a contour plot *and* the corresponding
+location on the colorbar. In this more common use case,
+`~matplotlib.colorbar.Colorbar.add_lines` can be used.
```

Comment:
`add_lines` requires a `ContourSet` so I don't understand this comment.  Either it need to be expanded, or removed.